src/Courses/CourseCreation/CourseEditPage.tsx

```diff
@@ -1,41 +1,73 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import EnterRightAnimWrapper from './EnterRightAnimWrapper';
 import TopicsList from '../TopicsList';
-import { Button, Col, Row, Accordion, Card } from 'react-bootstrap';
+import { Button, Col, Row, Accordion, Card, Modal } from 'react-bootstrap';
+import AxiosRequest from '../../Hooks/AxiosRequest';
+import { useParams } from 'react-router-dom';
+import TopicCreationModal from './TopicCreationModal';
+import _ from 'lodash';
+import { TopicObject } from '../CourseInterfaces';
 
 interface CourseEditPageProps {
 
 }
 
-const mock_topics = [
-    {topic_name: 'addition', topic_id: 1},
-    {topic_name: 'subtraction', topic_id: 2},
-    {topic_name: 'multiplication', topic_id: 3},
-    {topic_name: 'english', topic_id: 4}
-];
-const mock_units = [
-    {name: 'Unit 1', topics: mock_topics, unit_id: 1}, 
-    {name: 'Unit 2', topics: mock_topics, unit_id: 2},
-    {name: 'Unit 3', topics: mock_topics, unit_id: 3},
-];
-
-
 /**
  * This page requires an ICourseTemplate ID.
- * 
+ * Editing a Template Course means getting all details from that template, but saving
+ * to a new row.
  */
 export const CourseEditPage: React.FC<CourseEditPageProps> = () => {
+    const { courseId } = useParams();
+    const [course, setCourse] = useState<any>(null);
+    const [showTopicCreation, setShowTopicCreation] = useState<{show: boolean, unit: number}>({show: false, unit: -1});
+
+    useEffect(() => {
+        (async ()=>{
+            let course = await AxiosRequest.get(`/curriculum/${courseId}`);
+
+            console.log(course.data.data);
+            setCourse(course.data.data);
+        })();
+    }, []);
+
+
+    const callShowTopicCreation = (unit: number, e: any = null) => {
+        if (e != null) {
```

Comment:
why single equals?

Reply:
null and undefined I would guess, but does this also catch things like false?  Even though it should be fine I feel like we should keep up with good practice, doesn't linter yell about this?

Reply:
Using soft equals is fine where it's intentional. In this case, I want any falsy value to fail. e should be an event, anyhow.